File: LongestRepeatingCharacterReplacement.java
class Solution {
    public int characterReplacement(String s, int k) {
        char[] sarray = s.toCharArray();
        HashMap<Character, Integer> map = new HashMap<>();
        int left = 0;
        int right = 0;
        int totalCount = 0;
        int prev = -1;
        int max = 0;
        while(right < sarray.length){
            if(!map.containsKey(sarray[right])){
                map.put(sarray[right], 1);
            }
            else{
                if(right != prev){
                    int temp = map.get(sarray[right])+1;
                    map.put(sarray[right], temp);
                }
            }
            prev = right;
            max = Math.max(max, map.get(sarray[right]));
            if((right-left+1)-max <= k){
                totalCount = Math.max(totalCount, right-left+1);
                right++;
            }
            else{                  
                int temp = map.get(sarray[left])-1;
                map.put(sarray[left], temp);
                left++;
            }
        }
        return totalCount;
    }
}

File: LongestRepeatingCharacterReplacement.py
class Solution(object):
    def characterReplacement(self, s, k):
        """
        :type s: str
        :type k: int
        :rtype: int
        """
        left = 0
        letters = {}
        maxcount = 0
        for right in range(len(s)):
            letters[s[right]] = 1 + letters.get(s[right], 0)
            while(right - left + 1 - max(letters.values()) > k):
                letters[s[left]] = letters[s[left]] - 1
                left += 1
            maxcount = max(maxcount, right - left + 1)
        return maxcount
        
        